sdk/lib/opentelemetry/sdk/trace/span.rb

```diff
@@ -343,14 +343,12 @@ def trim_span_attributes(attrs)
 
           excess = attrs.size - @span_limits.attribute_count_limit
           excess.times { attrs.shift } if excess.positive?
-          truncate_attribute_values(attrs)
+          truncate_attribute_values(attrs, @span_limits.attribute_length_limit)
           nil
         end
 
-        def truncate_attribute_values(attrs)
+        def truncate_attribute_values(attrs, attribute_length_limit)
           return EMPTY_ATTRIBUTES if attrs.nil?
-
-          attribute_length_limit = @span_limits.attribute_length_limit
```

Comment:
I believe this should/could be considered a breaking change. However I don't know how flexible that is when it was doing the incorrect thing previously.  This line was applying span specific limits against the event attribute length.

Reply:
Yeah, for some definition of breaking 😂. Technically we broke the API defined by environment variables. IMO a minor version bump with a note in the changelog would be fine.

Reply:
I stand by my comment on May 9, which was approx 300 years ago. I think a minor version bump is fine!